Library/Homebrew/formulary.rb

```diff
@@ -226,6 +228,13 @@ def self.loader_for(ref)
       return AliasLoader.new(possible_alias)
     end
 
+    if newref = FORMULA_RENAMES[ref]
+      formula_with_that_oldname = core_path(newref)
+      if formula_with_that_oldname.file?
+        return FormulaLoader.new(newref, formula_with_that_oldname)
+      end
+    end
```

Comment:
If we put it after tap formula resolution, then `resolved_formula` returns a formula from tap if core installed and core renamed:

* Core `ack` installed
* Core `ack` renamed
* Tap `ack` available
* `brew migrate ack` outputs: ack doesn't replace any formula (it happens because tap ack is instantiated).